RFC.md

```diff
@@ -1,19 +1,171 @@
+})
+
+// Events and payload data are not inferred.
+Hub.listen(channel, ({ payload }) => {
+  switch (payload.event) {
+    case "signInFailure":
+      const data = payload.data;
+      break;
+  }
+});
+```
+
+**Amplify v6 (`aws-amplify@6`)**
+
+```Typescript
+import {Hub} from "aws-amplify"
+
+const channel = "auth"
+const signInFailureEvent = "signInFailure"
+const signInFailureData = new AuthError("sign-in failed")
+
+// By default, the dispatch API will provide a list of all Amplify supported channels.
+// Addionally, it shows all the events and data associated to a given channel.
+Hub.dispatch(channel,
+ {
+  event: signInFailureEvent ,
+  data: "string" // it errors as "string" is not AuthError
+} )
+
+// It listens for events associated to a channel and infers the right payload data on the switch statement.
+Hub.listen(channel, ({ payload }) => {
+	switch (payload.event) { // event - signIn | signInFailure | signUp | signUpFailure | ...etc
+		case "signInFailure":
+			const data = payload.data; // data - AuthError
+			break;
+	}
+});
+```
+
+## TypeScript support for custom Hub channels
+
+We are improving **_DX_** by adding strict type support to custom Hub channels, events, and payloads.
+
+**Amplify v5 (`aws-amplify@5`)**
+
+```Typescript
+const customChannel = "custom_channel"
+const customEvent = "custom_event"
+const customData = "custom_data"
+
+Hub.dispatch(channel, {
+  event: customEvent,
+  data: customData
+})
+
+Hub.listen(channel, ({ payload }) => {
+  switch (payload.event) {
+    case customEvent:
+      const data = payload.data;
+      break;
+  }
+});
+```
+
+**Amplify v6 (`aws-amplify@6`)**
+
+```Typescript
+type customChannel = "custom_channel"
+
+// Each key in the map represents a payload event and the key value is the data type for that event.
+// Note: If an event is assigned the null type, the payload object will not contain a data key.
+type customEventDataMap = {
+  A: number;
+  B: string;
+  C: null;
+  D: Object
+}
+
+Hub.dispatch<customChannel, customEventDataMap >("custom_channel", {
+  event: "A" , // Union type of all possible events A | B | C | D
+  data: 42
+})
+
+Hub.listen<customChannel, customEventDataMap>("custom_channel", ({payload}) => {
+  switch(payload.event) {
+    case "A":
+      payload.data; // data - number
+      break
+    case "B":
+      payload.data // data - string
+      break
+    case "C":
+      payload.data // it errors, there is no a data key on the payload object
+      break
+    case "D":
+      payload.data // data - Object
+      break
+  }
+})
+```
+
+## TypeScript support for Amplify Configuration
+
+To help customers configure Amplify categories, we are introducing type support for the `Amplify.configure` API. The examples below demonstrate an `Auth` configuration.
+
+**Amplify v5 (`aws-amplify@5`)**
+
+```Typescript
+const authConfig = {
+  userPoolId: "us-east-1_0yqxxHm5q",
+  userPoolClientId: "3keodiqtm52nhh2ls0vQfs5v1q",
+  signUpVerificationMethod: "code"
+}
```

Comment:
Why not highlight the fact that is not type and you can make typo mistakes?

Reply:
yeah I'm gonna add screen shoots showing that